packages/mui-material/src/ListSubheader/ListSubheader.js

```diff
@@ -100,6 +100,8 @@ const ListSubheader = React.forwardRef(function ListSubheader(inProps, ref) {
   );
 });
 
+ListSubheader.muiSkipListHighlight = true;
```

Comment:
There is already a detection mechanism implemented that skips non focusable items in the list that seems to work reliably otherwise keyboard interaction wouldn't work correctly (see grouping in the [docs demo](https://mui.com/material-ui/react-select/#grouping)). I suggest to use the same detection here instead of introducing a new static prop that adds a fairly important implementation detail to all the implementers? The other one works out of the box without anyone thinking or doing anything about it. It just skips non focusable items. It **also works** if you disable some otherwise focusable `MenuItem`.

Reply:
It doesn't affect all implementers, just those who wrap the ListSubheader in a custom component.

The existing keyboard navigation behavior relies on the DOM nodes already being constructed, as it checks the existence of `aria-disabled` and `tabindex` attributes and focuses the right item.

The initial highlight works differently as it has to overwrite the tabIndex **prop** of the first focusable item. Overwriting the `tabindex` attribute won't work here, as it would conflict with the prop.

There certainly could be a way to redesign how this logic works, but I'm not eager to rewrite the Select. We've already implemented the SelectUnstyled, which will be used to power the Material UI's Select in the upcoming version. It is free from this bug (and many others).

Reply:
The `tabIndex` solution seems much better and universal than this prop.